### PR TITLE
Add temperature and top-p

### PR DESCRIPTION
**Pull request by RyanMarten**
Doing this since Negin requires it for UnnaturalInstructions. 

Also to get an idea of the best way to do #62 and #74. 

This is just a stop gap before I get to those two issues.

### CODE REVIEW

File: src/bespokelabs/curator/prompter/prompter.py [hunk 1]
```diff
@@ -50,6 +50,8 @@ def __init__(
         response_format: Optional[Type[BaseModel]] = None,
         batch: bool = False,
         batch_size: Optional[int] = None,
+        temperature: float = 1.0,
+        top_p: float = 1.0,
```
Comment (madiator) [inline src/bespokelabs/curator/prompter/prompter.py:54]: Are these the right defaults to use?
Should it not be None?
https://docs.litellm.ai/docs/completion/input:
```
    temperature: Optional[float] = None,
```
If it is None, I assume litellm doesn't pass the temperature and the provider will the default that they have configured.

**Review by CharlieJCJ — APPROVED** [review]
LGTM!

File: src/bespokelabs/curator/prompter/prompter.py (hunk 1 above)
Comment (RyanMarten) [reply src/bespokelabs/curator/prompter/prompter.py:54]: Fixing this!

**Comment by RyanMarten** [comment]
Test with 
```
from bespokelabs.curator import Prompter
from datasets import Dataset

dataset = Dataset.from_dict({"prompt": ["write me a poem"] * 10})

prompter = Prompter(
    prompt_func=lambda row: row["prompt"],
    model_name="gpt-4o-mini",
    response_format=None,
    temperature=TEMP,
)

dataset = prompter(dataset)
print(dataset.to_pandas())
```

Temperature `0.0`
```
0  In the hush of dawn's embrace,  \nWhere whispe...
1  In the hush of dawn's embrace,  \nWhere whispe...
2  In the hush of dawn's embrace,  \nWhere whispe...
3  In the hush of dawn's embrace,  \nWhere whispe...
4  In the hush of dawn's embrace,  \nWhere whispe...
5  In the hush of dawn's embrace,  \nWhere whispe...
6  In the hush of dawn's embrace,  \nWhere whispe...
7  In the hush of dawn's embrace,  \nWhere whispe...
8  In the hush of dawn's embrace,  \nWhere whispe...
9  In the hush of dawn's embrace,  \nWhere whispe...
```

Temperature `1.5`
```
0  In the hush of morning's light,  \nWhere whisp...
1  In the quiet hush of dawning light,  \nThe wor...
2  In a quiet grove where violets dance,  \nBenea...
3  In twilight’s hush, when shadows play,  \nThe ...
4  In a garden where the blossoms sing,  \nBeneat...
5  Beneath the sky, in twilight hues,  \nA whispe...
6  In the hush of dawn's embrace,  \nA gentle whi...
7  In the garden where the wildflowers sway,  \nU...
8  In a quiet glade where shadows dwell,  \nWhisp...
9  In the quiet of the dawn, when shadows blend w...
```

**Pull request merged**